packages/bot-web-ui/src/stores/load-modal-store.js

```diff
@@ -0,0 +1,248 @@
+import {
+    observable,
+    action,
+}                         from 'mobx';
+import {
+    load,
+    config,
+    save_types,
+    getRecentFiles,
+}                         from '@deriv/bot-skeleton';
+
+export default class LoadModalStore {
+
+    @observable is_load_modal_open = false;
+    @observable active_index = 0;
+    @observable recent_files = [];
+    @observable selected_file = '';
+    @observable explaination_expand = false;
+    @observable loaded_local_file = null;
+    recent_workspace;
+    local_workspace;
+    drop_zone;
+
+    constructor(root_store) {
+        this.root_store = root_store;
+    }
+
+    @action.bound
+    toggleLoadModal() {
+        this.is_load_modal_open = !this.is_load_modal_open;
+
+        if (this.is_load_modal_open) {
+            this.recent_files = getRecentFiles() || [];
+        }
+    }
+
+    @action.bound
+    setActiveTabIndex(index) {
```

Comment:
Could you add comments to each condition in this function that explain what's happening?